src/hooks/useDashboards.tsx

```diff
@@ -145,34 +144,52 @@ export const useDashboardsQuery = (opts: { updateTimeRange?: (dashboard: Dashboa
 	};
 };
 
-export const useTileQuery = (opts?: { tileId?: string; onSuccess?: (data: TileQueryResponse) => void }) => {
+export const useTileQuery = (opts: {
+	tileId?: string;
+	query: string;
+	startTime: Date;
+	endTime: Date;
+	onSuccess?: (data: TileQueryResponse) => void;
+	enabled?: boolean;
+}) => {
 	const [, setDashboardsStore] = useDashboardsStore((_store) => null);
-	const { onSuccess } = opts || {};
-	const [fetchState, setFetchState] = useState<{
-		isLoading: boolean;
-		isError: null | boolean;
-		isSuccess: null | boolean;
-	}>({ isLoading: false, isError: null, isSuccess: null });
+	const { onSuccess, query, startTime, endTime, tileId, enabled = true } = opts;
 
-	const fetchTileData = useCallback(
-		async (queryOpts: TileQuery) => {
-			try {
-				setFetchState({ isLoading: true, isError: null, isSuccess: null });
-				const res = await getQueryData(queryOpts);
-				const tileData = _.isEmpty(res) ? { records: [], fields: [] } : res.data;
-				opts?.tileId && setDashboardsStore((store) => setTileData(store, opts.tileId || '', tileData));
-				opts?.onSuccess && opts.onSuccess(tileData);
-				setFetchState({ isLoading: false, isError: false, isSuccess: true });
-			} catch (e: any) {
-				setFetchState({ isLoading: false, isError: true, isSuccess: false });
-				notifyError({ message: _.isString(e.response.data) ? e.response.data : 'Unable to fetch tile data' });
-			}
+	const fetchTileData = async (queryOpts: TileQuery) => {
```

Comment:
Wrap it with `useCallback`.
And as the name suggests it should just fetch the tile data and return the response. And the success block inside the useQuery should handle setting the store. And move this fn out of this to any of the API files so that we can call this independently wherever we want.

Reply:
Have addressed this